wordpress-mobile/WordPress-Android
# Issue/13329 error handling

Parent #13329 

This PR adds logic to use the last step of the wizard "Complete" in both complete and error capacities.
- Removed "Error" UiState and refactored"Content" UiState in the viewmodels, as only one state is needed since everything is a list of items
- Updated `BackupDownloadCompleteStateListItemBuilder` to include building out a JetpackItemList for error
- Updated `GetBackupDownloadStatusUseCase` to check for network connectivity after each delay to fix a potential never ending loop issue.
- Send errors to Complete view as outlined in the following image

<img src="https://user-images.githubusercontent.com/506707/104215898-69b66f00-5407-11eb-86a2-fee981fb475c.png" width="300" alt="Screenshot">

**Notes**
- Please pay no attention to styles
- Complete success buttons are not hooked up yet

**To test:**
- Launch the app with BackupDownloadConfig flag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Download Backup menu item
- Tap the Create downloadable file button
- When the view changes to progress, quickly turn on airplane mode
- Note that the view changes to "Download Failed"
- Click the Done button to close the wizard
- Don't forget to turn airplane mode back on after the test is over

PR submission checklist:
